File: backend/api_gateway/src/main/java/com/ticketflow/api_gateway/service/seeders/CinemaHallsSeeder.java
package com.ticketflow.api_gateway.service.seeders;

import java.util.List;

import com.ticketflow.api_gateway.models.movie_service.CinemaHall;
import com.ticketflow.api_gateway.proxy.movie.cinema_halls_api.CinemaHallsApiProxy;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class CinemaHallsSeeder {
    private List<CinemaHall> dataToSeed = List.of(
        new CinemaHall("SuperLux", "Kyiv", 5, 5),
        new CinemaHall("JustRelax", "Kyiv", 5, 5),
        new CinemaHall("RedHall", "Lviv", 5, 5)
    );

    private CinemaHallsApiProxy cinemaHallsApiProxy;
    
    @Autowired
    public CinemaHallsSeeder(CinemaHallsApiProxy cinemaHallsApiProxy) {
        this.cinemaHallsApiProxy = cinemaHallsApiProxy;
    }

    public void seed() {
        if (cinemaHallsApiProxy.getAll().isEmpty()) {
            dataToSeed.forEach(cinemaHallsApiProxy::add);
        }
    }
}

File: backend/dotnet/ticketflow/TicketFlow.ApiGateway/StartupServices/Seeders/CinemaHallsSeeder.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using TicketFlow.MovieService.Client.Extensibility.Entities;
using TicketFlow.MovieService.Client.Extensibility.Models.CinemaHallModels;
using TicketFlow.MovieService.Client.Extensibility.Proxies;

namespace TicketFlow.ApiGateway.StartupServices.Seeders
{
    internal class CinemaHallsSeeder : ICinemaHallsSeeder
    {
        private readonly ICinemaHallApiProxy cinemaHallApiProxy;

        private readonly IReadOnlyCollection<CinemaHallCreationModel> cinemaHallsToSeed = new[]
        {
            new CinemaHallCreationModel("SuperLux", "Kyiv", 5, 5),
            new CinemaHallCreationModel("JustRelax", "Kyiv", 5, 5),
            new CinemaHallCreationModel("RedHall", "Lviv", 5, 5)
        };

        public CinemaHallsSeeder(ICinemaHallApiProxy cinemaHallApiProxy)
        {
            this.cinemaHallApiProxy = cinemaHallApiProxy;
        }

        public async Task SeedAsync()
        {
            IReadOnlyCollection<ICinemaHall> storedCinemaHalls = await cinemaHallApiProxy.GetAllAsync();
            if (storedCinemaHalls.Count == 0)
            {
                foreach (var cinemaHall in cinemaHallsToSeed)
                {
                    await cinemaHallApiProxy.AddAsync(cinemaHall);
                }
            }
        }
    }
}